src/EnergyPlus/Coils/CoilCoolingDXCurveFitPerformance.cc

```diff
@@ -176,152 +187,151 @@ void CoilCoolingDXCurveFitPerformance::simulate(EnergyPlus::EnergyPlusData &stat
                                                 Real64 &PLR,
                                                 int &speedNum,
                                                 Real64 &speedRatio,
-                                                int &fanOpMode,
+                                                int const fanOpMode,
                                                 DataLoopNode::NodeData &condInletNode,
                                                 DataLoopNode::NodeData &condOutletNode,
+                                                bool const singleMode,
                                                 Real64 LoadSHR)
 {
     Real64 reportingConstant = DataHVACGlobals::TimeStepSys * DataGlobalConstants::SecInHour;
+    this->recoveredEnergyRate = 0.0;
+    this->NormalSHR = 0.0;
 
     if (useAlternateMode == DataHVACGlobals::coilSubcoolReheatMode) {
         Real64 totalCoolingRate;
         Real64 sensNorRate;
         Real64 sensSubRate;
         Real64 sensRehRate;
+        Real64 latRate;
         Real64 SysNorSHR;
         Real64 SysSubSHR;
         Real64 SysRehSHR;
-        Real64 minAirHumRat;
         Real64 HumRatNorOut;
         Real64 TempNorOut;
         Real64 EnthalpyNorOut;
         Real64 modeRatio;
 
-        this->recoveredEnergyRate = 0.0;
-        this->NormalSHR = 0.0;
-        this->calculate(state, this->normalMode, inletNode, outletNode, PLR, speedNum, speedRatio, fanOpMode, condInletNode, condOutletNode);
+        this->calculate(
+            state, this->normalMode, inletNode, outletNode, PLR, speedNum, speedRatio, fanOpMode, condInletNode, condOutletNode, singleMode);
 
-        //this->OperatingMode = 1;
-        totalCoolingRate = outletNode.MassFlowRate * (inletNode.Enthalpy - outletNode.Enthalpy);
-        minAirHumRat = min(inletNode.HumRat, outletNode.HumRat);
-        sensNorRate = outletNode.MassFlowRate *
-                      (Psychrometrics::PsyHFnTdbW(inletNode.Temp, minAirHumRat) - Psychrometrics::PsyHFnTdbW(outletNode.Temp, minAirHumRat));
+        // this->OperatingMode = 1;
+        CalcComponentSensibleLatentOutput(
+            outletNode.MassFlowRate, inletNode.Temp, inletNode.HumRat, outletNode.Temp, outletNode.HumRat, sensNorRate, latRate, totalCoolingRate);
         if (totalCoolingRate > 1.0E-10) {
             this->OperatingMode = 1;
             this->NormalSHR = sensNorRate / totalCoolingRate;
+            this->powerUse = this->normalMode.OpModePower;
+            this->RTF = this->normalMode.OpModeRTF;
+            this->wasteHeatRate = this->normalMode.OpModeWasteHeat;
         }
 
-        if (PLR == 0.0) return;
-        if (LoadSHR == 0.0) return;
+        if ((PLR != 0.0) && (LoadSHR != 0.0)) {
```

Comment:
Changed the if structure here so it still computes if one of these is zero but not both. New if indent makes for lots of visual diffs below.

Reply:
Hmm, I assume you tested this but it doesn't look like it does what you say. If PLR = 0, the PLR != 0 is false and when using && it doesn't matter what the other is, it won't execute. So it only executes when both are non-zero. Maybe that's what you expect?

false && false = false, PLR = 0 and LoadSHR = 0
false && true = false, PLR = 0 and LoadSHR > 0
true && false = false, PLR > 0 and LoadSHR = 0
true && true = true, PLR > 0 and LoadSHR > 0

Reply:
Oh, you're right. Looking at the whole file, this was added so that the crankcase heater and basin heater calcs will happen at the end of the function (instead of `return`ing when the load is zero for the subcool reheat mode). It's hard to see in the github diffs, but the basin heater and cranckcase heater were moved from `calculate` to `simulate`, You almost need to open before and after side-by-side to follow these changes.

Reply:
Now I'm really not sure about this logic and where these calcs were included. Crankcase and basin heaters are a function of OAT (and PLR for the calculation for crankcase) and should probably always be calculated. Just saying this so we make sure that logic is right.